src/lib.rs

```diff
@@ -106,6 +106,19 @@ use std::fmt;
 use std::error;
 use std::collections::HashMap;
 
+#[derive(Debug)]
+pub enum Trap {
```

Comment:
Oh its' pretty easy to document this from official spec?

Reply:
Yup! TBH I just forgot to add the docs :s